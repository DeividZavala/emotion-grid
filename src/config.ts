import { Grid, Breakpoints, Theme } from './types';

export const BREAKPOINTS: Array<Breakpoints> = ['xs', 'sm', 'md', 'lg', 'xl'];

function config(props?: Theme): Grid {
  const base = {
    container: {
      xs: 'full',
      sm: 'full',
      md: 'full',
      lg: 90,
      xl: 90,
    },
    padding: {
      xs: 1,
      sm: 1,
      md: 1.5,
      lg: 1.5,
      xl: 1.5,
    },
    columns: {
      xs: 4,
      sm: 8,
      md: 8,
      lg: 12,
      xl: 12,
    },
    gutter: {
      xs: 1,
      sm: 1,
      md: 1.875,
      lg: 1.875,
      xl: 1.875,
    },
    breakpoints: {
      xs: 0,
      sm: 0,
      md: 48,
      lg: 64,
      xl: 71.25,
    },
  };

  return { ...base, ...props };
}

export default config;
