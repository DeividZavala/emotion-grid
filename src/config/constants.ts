import { Types } from '../types';

export const BREAKPOINTS: Array<Types.Breakpoints> = [
  'xs',
  'sm',
  'md',
  'lg',
  'xl',
];

export const DEFAULT_CONFIG: Types.Grid = {
  container: {
    xs: 'full',
    sm: 'full',
    md: 45,
    lg: 60,
    xl: 71.25,
  },
  padding: {
    xs: 1.875,
    sm: 1.875,
    md: 1.875,
    lg: 1.875,
    xl: 1.875,
  },
  columns: {
    xs: 4,
    sm: 8,
    md: 12,
    lg: 12,
    xl: 12,
  },
  gutter: {
    xs: 1.875,
    sm: 1.875,
    md: 1.875,
    lg: 1.875,
    xl: 1.875,
  },
  breakpoints: {
    xs: 1,
    sm: 33.75,
    md: 48,
    lg: 64,
    xl: 71.25,
  },
  debug: {
    color: '#0470F5',
  },
};
