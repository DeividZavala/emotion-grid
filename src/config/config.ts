import { Types } from '../types';
import { DEFAULT_CONFIG } from './constants';

function isTheme(props: Types.ThemeProps): props is Types.Theme {
  return (props as { theme: Types.Theme }).theme === undefined;
}

function getTheme(props: Types.ThemeProps): Types.Theme {
  return isTheme(props) ? props : props.theme;
}

function config(props: Types.Theme = {}): Types.Grid {
  const theme = getTheme(props);

  return {
    ...DEFAULT_CONFIG,
    ...theme.grid,
  };
}

export default config;
