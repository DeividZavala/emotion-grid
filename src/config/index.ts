import * as constants from './constants';
export { default as config } from './config';

export { constants };
