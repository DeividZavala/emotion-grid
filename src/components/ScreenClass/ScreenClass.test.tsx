import React from 'react';
import mediaQuery from 'css-mediaquery';
import { render } from '@testing-library/react';

import { config } from '../../config';
import { ScreenClass } from '.';

const createMatchMedia = (width: number) => (query: string) => ({
  matches: mediaQuery.match(query, { width }),
});

describe('ScreenClass', () => {
  it.each(Object.values(config().breakpoints))(
    'should render with current screen %s',
    (breakpoint) => {
      // @ts-ignore
      window.matchMedia = createMatchMedia(breakpoint * 16);

      const { container } = render(
        <ScreenClass
          render={(screen) => {
            return <div>{screen}</div>;
          }}
        />
      );

      expect(container.firstChild).toMatchSnapshot();
    }
  );
});
