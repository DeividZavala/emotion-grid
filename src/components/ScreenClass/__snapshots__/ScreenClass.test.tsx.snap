// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScreenClass should render with current screen %s 1`] = `
<div>
  xl
</div>
`;
